### DISCUSSION TITLE
Filmon Script Player supported ?

### DISCUSSION THREAD

**Discussion by Sphinxroot** [iptv-org/community#95] [Show and tell]
i founded a way to play filmon stream without cut after 30sec of preview

in kodi it work like a charm  i need to know for other player if its good or not

this is an exemple :
#EXTINF:-1 tvg-logo="https://i.imgur.com/ct56rK2.png" group-title="",5usa (320p)
https://nowontv.info/filmon.php?url=https://www.filmon.com/api-v2/channel/845?protocol=hls

you  change then the channel number  from any link from filmon and it work but it dont give more than 320p

#EXTINF:-1 tvg-logo="" group-title="",FoodNetwork
https://edge-1180-ch-gv.filmon.com/live/20.low.stream/playlist.m3u8+token will give you 30sec

#EXTINF:-1 tvg-logo="" group-title="",FoodNetwork (320p)
https://nowontv.info/filmon.php?url=https://www.filmon.com/api-v2/channel/20?protocol=hls

https://i.imgur.com/1Sdz7H6.png

**Comment by Dum4G** [comment]
Hangs on after a minute or two on mobile VLC

**Comment by Sphinxroot** [comment]
ok so it wont be universal for all player   or its the network caching since kodi u can edit it globaly

i dont know if adding

#EXTVLCOPT:network-caching=1000    (changing it to fit it could be the clue)

into it will do the trick

**Reply by rlaphoenix** [reply]
I can confirm `#EXTVLCOPT:network-caching=1000` being injected to the m3u8 is **required** for VLC to work continuously.
This does mean the streams do work, just it depends on the player.

EDIT: You can either inject that setting, or change it in VLC Settings globally.

I also noticed you did not give information about the filmon.php script you mentioned in the original post. However, it doesn't seem to be doing anything other than grabbing the low M3U stream URL and feeding it back out. Nothing more.

If anyone is interested in getting the streams working on VLC, or want an alternative to that PHP URL for Kodi or such, then take a look at: https://gist.github.com/rlaphoenix/e18170c950a917890823eb53777a15a0

You will need to host it yourself.

**I can also confirm both the low and high-quality streams work! No need to use the low stream!**

**Reply by rlaphoenix** [reply]
Small update, here's a script to convert all of FilmOn to a tvg M3U (logo, name, category included): https://gist.github.com/rlaphoenix/f10d3f39882011bdffee143bd504b149 it goes perfectly in hand with my previously linked script.

**Comment by BellezaEmporium** [comment]
Made a program that scraped links up to 9000 for FilmON TV

![image](https://user-images.githubusercontent.com/30985701/131028916-e76db353-63d3-4a5d-958f-3c868d6aae98.png)

Seems to work nicely.

**Reply by BellezaEmporium** [reply]
http://www.filmon.com/vr-streams/27.high/playlist.m3u8 - BBC News
http://www.filmon.com/vr-streams/316.high/playlist.m3u8 - Kix!
http://www.filmon.com/vr-streams/318.high/playlist.m3u8 - Tiny Pop
http://www.filmon.com/vr-streams/320.high/playlist.m3u8 - POP
http://www.filmon.com/vr-streams/1735.high/playlist.m3u8 - Perviy kanal Europa
http://www.filmon.com/vr-streams/1738.high/playlist.m3u8 - RTR Planeta
http://www.filmon.com/vr-streams/1744.high/playlist.m3u8 - Rossiya 24
http://www.filmon.com/vr-streams/1793.high/playlist.m3u8 - Eri TV
http://www.filmon.com/vr-streams/2798.high/playlist.m3u8 - Really TV
http://www.filmon.com/vr-streams/3029.high/playlist.m3u8 - Al Jazeera Documentary
http://www.filmon.com/vr-streams/3032.high/playlist.m3u8 - Al Jazeera Mubasher
http://www.filmon.com/vr-streams/3035.high/playlist.m3u8 - Al Jazeera Arabic
http://www.filmon.com/vr-streams/3166.high/playlist.m3u8 - BBC 1 Scotland
http://www.filmon.com/vr-streams/3386.high/playlist.m3u8 - Forces TV
http://www.filmon.com/vr-streams/3890.high/playlist.m3u8 - France 3 Rhône-Alpes
http://www.filmon.com/vr-streams/3893.high/playlist.m3u8 - France 3 Alpes
http://www.filmon.com/vr-streams/4121.high/playlist.m3u8 - Sport 1
http://www.filmon.com/vr-streams/4136.high/playlist.m3u8 - 360 Experimental
http://www.filmon.com/vr-streams/4151.high/playlist.m3u8 - Radio Chico
http://www.filmon.com/vr-streams/4721.high/playlist.m3u8 - Sophia TV
http://www.filmon.com/vr-streams/4724.high/playlist.m3u8 - Die neue Zeit TV
http://www.filmon.com/vr-streams/6031.high/playlist.m3u8 - R9 Oesterreich
http://www.filmon.com/vr-streams/6070.high/playlist.m3u8 - Shockya
http://www.filmon.com/vr-streams/6076.high/playlist.m3u8 - Alki Live
http://www.filmon.com/vr-streams/6131.high/playlist.m3u8 - Gurvey's Law
http://www.filmon.com/vr-streams/6140.high/playlist.m3u8 - Malibu CCTV 1
http://www.filmon.com/vr-streams/6149.high/playlist.m3u8 - Hologram-Icons
http://www.filmon.com/vr-streams/6152.high/playlist.m3u8 - Erotic Sensations (XXX)
http://www.filmon.com/vr-streams/6155.high/playlist.m3u8 - there's some homosexual porn in this, but can't have any info from the API, "not available in my region"
http://www.filmon.com/vr-streams/6158.high/playlist.m3u8 - Fetish Factory (XXX) (not quite sure about this name, lots of shemale content in this one)
http://www.filmon.com/vr-streams/6191.high/playlist.m3u8 - Sophia TV ES
http://www.filmon.com/vr-streams/6194.high/playlist.m3u8 - Sophia TV IT
http://www.filmon.com/vr-streams/6197.high/playlist.m3u8 - Sophia TV FR

**Reply by rlaphoenix** [reply]
The `vr-streams` dont seem valid at all, not anymore at least.

**Comment by Dum4G** [comment]
Ok, so I tried on desktop with and without network-caching=1000. Still isn't a permanent solution. Somehow it hangs when ads start and before it finishes

**Reply by rlaphoenix** [reply]
I just want to say, as I said in a different comment, network-caching=1000 is required for VLC.
But, your comment about hanging on ads is invalid. Its actually normal for them to seem to freeze for even a while in between. Its common across various UK channels to do that for whatever reason and is not a freeze by the stream, its the actual channel doing that.
Its possible you assumed it was the stream freezing and proceeded to stop VLC, while it was actually streaming.

Using my m3u proxy here: https://gist.github.com/rlaphoenix/e18170c950a917890823eb53777a15a0 I watched during a show, and throughout the credits both start and end did in fact "freeze" for a few seconds, but its normal. It continued to play after the credits. As I'm writing this I'm watching Black'ish on E4 `high` stream for 15 minutes total now.

**Comment by jkemp814** [comment]
@LaneSh4d0w Made a program that scraped links up to 9000 for FilmON TV

They all work on this end.

**Reply by BellezaEmporium** [reply]
Perfect !

**Comment by BellezaEmporium** [comment]
Tweaked my program a bit, does it onto two parts : Link verification (only analyses links) and then contacts Filmon's API to get information, based on my location

For now, it has found far more links that what I would have expected.

**Reply by BellezaEmporium** [reply]
27 - BBC News
316 - Kix!
318 - Tiny Pop
320 - POP
1684 - Acort Horror Network
1729 - ARTE Français
1735 - Perviy kanal Europa
1738 - RTR Planeta
1741 - MTA TV (not registered on FilmON, weird.)
1744 - Rossiya 24
1793 - Eri.TV (Ethiopia ?)
1915 - A religious program right now, can't say what's the name of it
2798 - Really
3029 - Al Jazeera Documentary
3032 - Al Jazeera Mubasher
3035 - Al Jazeera Arabic
3166 - BBC 1 Scotland
3386 - Forces TV
3482 - Aaj Tak
3887 - franceinfo
3890 - France 3 Rhône-Alpes
3893 - France 3 Alpes
4118 - CGTN Français
4121 - SPORT1
4136 - 360 Experimental
4151 - Radio Chico
4721 - Sophia TV
4724 - Die neue Zeit TV
4727 - HSE24 HD
4730 - HSE24 Extra
5971 - El Watania TV
6046 - TBN Russia
6070 - Shockya
6131 - Gurvey's Law
6140 - Malibu CCTV 1
6149 - Hologram-Icons
6191 - Sophia TV ES
6194 - Sophia TV IT
6197 - Sophia TV FR
6200 - 1776
6203 - Kennedy-Channel
